documentation/workspace_manager.tex

```diff
@@ -225,11 +231,13 @@ \subsection{Program configuration}
 The workspace is the most important implementation of the \TT{parse\_library\_provider} interface. It provides libraries based on the processor groups configuration described in \cref{libs_config}.
 
 %******************** Diagnosable *************************
-\sectionSrc{Diagnosable}
+\sectionSrc{Diagnostics}
 {parser\_library/src/diagnosable.h,parser\_library/src/diagnosable\_impl.h,parser\_library/src/diagnostic.h}
 A diagnostic is used to indicate a problem with source files, such as a compiler error or a warning. Some diagnostics are created in almost every component of the parser when it finds a problem with a source code. Diagnostics are also used in workspace to indicate problems with configuration files. After each parsing, we need to collect all the diagnostics from all the instances of all the components and pass them to the language server.
 
-To tackle this problem, there is interface \emph{diagnosable}. Its functionality is simple, it is used to add diagnostics, show his own and collect them from other diagnosable members. Each component that implements the interface is required to collect diagnostics from diagnosable objects it owns. So in the result, the diagnosable components are organised in a tree and with one call of \TT{collect\_diags} from the root of the tree, it is possible to get all diagnostics that were created since last such call.
+The components capable of collecting the diagnostics are organized in a tree where the root is the workspace manager. Starting from the root, each component collects the diagnostics of those children that are again capable of collecting or generating diagnostics.
```

Comment:
..in a tree in which..? Neviem ci where sa takto moze pouzit

Reply:
Podla mna aj aj, ak chces tak si to zmen